ethcore/src/client/client.rs

```diff
@@ -2112,10 +2113,15 @@ impl BlockChainClient for Client {
 
 	fn transact_contract(&self, address: Address, data: Bytes) -> Result<(), transaction::Error> {
 		let authoring_params = self.importer.miner.authoring_params();
+		let service_transaction_checker = ServiceTransactionChecker::default();
+		let gas_price = match service_transaction_checker.check_address(self, address) {
```

Comment:
`address` is contract address you try to call, not the sender (which should be checked here).